packages/runner/src/scheduler.ts

```diff
@@ -38,7 +38,7 @@ const MAX_ITERATIONS_PER_RUN = 100;
 
 export class Scheduler implements IScheduler {
   private pending = new Set<Action>();
-  private eventQueue: (() => void)[] = [];
+  private eventQueue: (() => any)[] = [];
```

Comment:
These functions do *not* return `void`, and their return values are meaningful to the system.